sdk/lib/atl/atlimage.h

```diff
@@ -1024,96 +1023,143 @@ class CImage
         return RGB(quad.rgbRed, quad.rgbGreen, quad.rgbBlue);
     }
 
-    struct EXTENSION_ENTRY
-    {
-        LPCWSTR pszExt;
-        GUID guid;
-    };
-
-    const GUID *FileTypeFromExtension(LPCWSTR pszExt) const
+    static CLSID
+    FindCodecForExtension(LPCTSTR dotext, const Gdiplus::ImageCodecInfo *pCodecs, UINT nCodecs)
     {
-        static const EXTENSION_ENTRY table[] =
-        {
-            {L".jpg", Gdiplus::ImageFormatJPEG},
-            {L".png", Gdiplus::ImageFormatPNG},
-            {L".bmp", Gdiplus::ImageFormatBMP},
-            {L".gif", Gdiplus::ImageFormatGIF},
-            {L".tif", Gdiplus::ImageFormatTIFF},
-            {L".jpeg", Gdiplus::ImageFormatJPEG},
-            {L".jpe", Gdiplus::ImageFormatJPEG},
-            {L".jfif", Gdiplus::ImageFormatJPEG},
-            {L".dib", Gdiplus::ImageFormatBMP},
-            {L".rle", Gdiplus::ImageFormatBMP},
-            {L".tiff", Gdiplus::ImageFormatTIFF}
-        };
-        const size_t count = _countof(table);
-        for (size_t i = 0; i < count; ++i)
+        for (UINT i = 0; i < nCodecs; ++i)
         {
-            if (::lstrcmpiW(table[i].pszExt, pszExt) == 0)
-                return &table[i].guid;
+            CString strSpecs(pCodecs[i].FilenameExtension);
+            int i0 = 0, i1;
```

Comment:
i0 and i1 are not descriptive. i1 --> ichSep ? (the ; separator)

Reply:
Renamed.